test/test_utils/Mocks.h

```diff
@@ -79,21 +80,29 @@ class MockSubscriber : public Subscriber<T> {
   /**
    * Block the current thread until either onSuccess or onError is called.
    */
-  void awaitTerminalEvent() {
+  void awaitTerminalEvent(int waitForMs = 1000) {
```

Comment:
Can the argument be `std::chrono::milliseconds`?  Don't need the `Ms` variable suffix then, and can do things like pass in `std::chrono::seconds(1)` instead of 1000.